backend/tracim_backend/lib/core/storage.py

```diff
@@ -0,0 +1,308 @@
+from tracim_backend.exceptions import PageOfPreviewNotFound
+from tracim_backend.exceptions import PreviewGeneratorPassthroughError
+from tracim_backend.exceptions import TracimUnavailablePreviewType
+from tracim_backend.exceptions import UnavailablePreview
+from tracim_backend.lib.utils.logger import logger
+
+TEMPORARY_PREFIX = "tracim-revision-content"
+
+
+class StorageLib:
+    """
+    Helper class to handle more easily usage with depot stored file
+    """
+
+    def __init__(self, app_config: CFG) -> None:
+        self.app_config = app_config
+        self.uploaded_file_depot = DepotManager.get(
+            app_config.UPLOADED_FILES__STORAGE__STORAGE_NAME
+        )
+        self.preview_manager = PreviewManager(app_config.PREVIEW_CACHE_DIR, create_folder=True)
+
+    def _get_depot_file(self, depot_file) -> StoredFile:
+        try:
+            return self.uploaded_file_depot.get(depot_file)
+        except IOError as exc:
+            logger.warning(
+                self, "Unable to get content filepath, depot is corrupted", exc_info=True
+            )
+            raise CannotGetDepotFileDepotCorrupted(
+                "depot file {} is not accessible, depot seems corrupted".format(depot_file.file_id)
+            ) from exc
+
+    def get_raw_file(
+        self,
+        depot_file: UploadedFile,
+        filename: str,
+        default_filename: str,
+        force_download: bool = None,
+        last_modified: datetime = None,
+    ) -> HapicFile:
+        file = self._get_depot_file(depot_file)
+        # INFO - G.M - 2019-08-08 - use given filename in all case but none or
+        # "raw", where filename returned will be a custom one.
+        if not filename or filename == "raw":
+            filename = default_filename
+        return HapicFile(
+            file_object=file,
+            mimetype=file.content_type,
+            filename=filename,
+            as_attachment=force_download,
+            content_length=file.content_length,
+            last_modified=last_modified or None,
+        )
+
+    def get_filepath(
+        self,
+        depot_file: UploadedFile,
+        file_extension: str = "",
+        temporary_prefix: str = TEMPORARY_PREFIX,
+    ) -> typing.Generator[str, None, None]:
+        depot_stored_file = self._get_depot_file(depot_file)  # type: StoredFile
+        if self.app_config.UPLOADED_FILES__STORAGE__STORAGE_TYPE == DepotFileStorageType.LOCAL.slug:
+            yield from self._get_valid_content_filepath_legacy(depot_stored_file)
+        else:
+            yield from self._get_valid_content_filepath(
+                depot_stored_file, file_extension=file_extension, prefix=temporary_prefix
+            )
+
+    @contextmanager
+    def preview_generator_filepath_context(
+        self, depot_file: UploadedFile, original_file_extension: str
+    ) -> typing.Generator[str, None, None]:
+        """
+
+        :param depot_file:
```

Comment:
Could you add some documentation? The goal is unclear to me.

Reply:
done.